### ISSUE TITLE
proper content warnings and more extensive sensitive content

### ISSUE THREAD

**Issue by hoodiek** [tootsuite/mastodon#239]
this is not saying that the current stuff is inadequate, it is merely stating that much more could be done. for example, talk of death, and suicide, can very easily trigger such people who might be wary of them in the first place. as a result, not only should images be able to be made able to be spoilered. i would suggest that if you click the sensitive content toggle, a text box could appear with a simple content warning of sorts. for example "CW food" can be placed in a textbox, and the first few words of the post would have that added, removing from the max character limit. this would make the whole toot act like the current sensitive image filter does, and better allow sensitive toots in general. thanks for listening.

EDIT: a suggestion to avoid the CW necessity is, sensitive content tweets would mute all except #hashtags and thus be able to show stuff selectively

**Comment by Gargron** [comment]
I know some people might have an issue with TW/CW so let me just add that it'd also be perfect for hiding spoilers, something Twitter/FB have never done. I'll see if I can do something about this

**Comment by hoodiek** [comment]
yep! i'd really appreciate this addition! thank you so much for caring and listening, @Gargron

**Comment by Xkeeper0** [comment]
Perhaps having the "Sensitive content" box allow for a short free-form description of what's sensitive about it, while possibly marking the main text invisible instead of just the picture.

**Comment by marrus-sh** [comment]
Was just thinking about this and definitely favor having a separate content description textbox over just muting everything except hashtags. Take the example post

> omg there was a giant tarantula in my bathtub today `#spiders`

By clicking on `#spiders`, which the author is required to include in order to indicate the manner of sensitive content, users can then view all spiders-related tweets in one place. While this isn't probably a big deal with arachnids, it would make it incredibly easy for people to target, say, abuse survivors simply by searching the relevant tags.

**TL:DR;** using something designed as a search feature for a sensitive content marker might not be the best idea? Making it a separate piece of data would probably be better.

**Comment by NireBryce** [comment]
A thing i would like is a second type of mute, where instead of ignoring the tweet, it just spoilers it as if it was sensitive content

so I could say, block anything containing "CW: spiders" and it would just spoiler everything but the keyword, or maybe anything on the same line

**Comment by hoodiek** [comment]
i know this thread is very old but i think it is a strong tool to give users. i think the best middle-ground would be a click to show/hide sensitive content, and it being expanded to work like this:

sensitive content toggle would be visible for all posts, and anything in a (cw[...]) type tag would be visible. clicking on the parenthesesed thing would allow you to hide or show the content

this also allows people to mute the things as well as making spoilers for them
